### PR TITLE
Introduces common DosomethingUserRemote abstract class.

### PR DESCRIPTION
**Pull request by sergiitk**
#### What's this PR do?
- Splits `dosomething_uk_sso_user.inc` to fields common and UK-specific fields
- Creates new abstract class of a remote user in `dosomething_user` to extend other remote users from (e.g. Canada)
#### Testing

screen shot 2014-12-03 at 12 40 55 am
image

### CODE REVIEW

File: lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc [hunk 1]
```diff
@@ -74,6 +68,13 @@ class DosomethingUkRegisterController implements ExternalAuthRegisterController
       );
     }
 
+    // Determines whether the user agreed to receive e-mail newsletter.
+    if (!empty($values['field_opt_in_email'][LANGUAGE_NONE][0]['value'])) {
+      $user_data['contactable'] = TRUE;
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc:73]: A user potentially could opt-out of email, and/or SMS.  Would it make sense to store this as just `opt_in_email` to keep it consistent?

File: lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc [hunk 2]
```diff
@@ -63,7 +57,7 @@ class DosomethingUkRegisterController implements ExternalAuthRegisterController
     // @todo: use normal value when the international phone validator is ready.
     if (!empty($form_state['input']['field_mobile'][LANGUAGE_NONE][0]['value'])) {
       $phone = $form_state['input']['field_mobile'][LANGUAGE_NONE][0]['value'];
-      $this->remote_account->setPhoneNumber($phone);
+      $user_data['phone_number'] = $phone;
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc:60]: Any reason to rename as `'phone_number'`vs keeping as `mobile` to keep things consistent?

File: lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_sso_user.inc [hunk 3]
```diff
@@ -152,43 +82,51 @@ class DosomethingUkSsoUser {
...
-      $data['dob']
+      $data['first_name'],
+      $data['dob'],
+      (isset($data['phone_number']) ? $data['phone_number'] : FALSE)
     );
+
+    // Set UK-specific fields.
+    $instance->last_name   = $data['last_name'];
+    $instance->postcode    = $data['postcode'];
+    $instance->screen_name = strtolower($instance->getFirstName()) . time();
```
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_sso_user.inc]: Potentially call this username? I'm just not crazy about having properties that seem to match with the Drupal User, but are named differently... but definitely open to hear the reasoning...

File: lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc (hunk 1 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc:73]: 1. SMS switch is still not implemented from the UK side
2. `contactable` Is actual name of field in the remote user. The idea is to represent the structure of the remote user.
   image

**Comment by aaronschachter** [comment]
Code looks okay, just questions around naming conventions.

File: lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc (hunk 2 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_uk/includes/dosomething_uk_register_controller.inc:60]: Same as in `contactable`, `phone_number` is the actual name of the remote field.

**Comment by sergiitk** [comment]
Discussed with @aaronschachter: revisit common fields in `DosomethingUserRemote`.

**Comment by sergiitk** [comment]
image
image

@aaronschachter could you please re-review?

**Comment by aaronschachter** [comment]
:+1:

**Pull request merged**